### PR TITLE
Reformat api.md

### LINKED ISSUE #8178: API doc reformatted
The file is reformatted, but not changed? I can't tell. Please move the reformatting to a different PR.

_Originally posted by @fniessink in https://github.com/ICTU/quality-time/pull/8163#discussion_r1505825721_

### PR DESCRIPTION
**Pull request by wkoot**
Closes #8178

### CODE REVIEW

File: docs/src/api.md [hunk 1]
```diff
@@ -68,30 +72,52 @@ The `search_query` is not included in the error response if parsing the search q
...
+To hide metrics that do not need any action, set the `metrics_to_hide` parameter to `no_action_needed`,
+i.e. `https://www.quality-time.example.org/api/v3/report/<report_uuid>/pdf?metrics_to_hide=no_action_needed`.
+To hide all metrics and only export the report dashboard to PDF, set the `metrics_to_hide` parameter to `all`.
 
-To hide columns from the report, set the `hidden_columns` parameter, for example `https://www.quality-time.example.org/api/v3/report/<report_uuid>/pdf?hidden_columns=target,comment`. Possible options are `trend`, `status`, `measurement`, `target`, `source`, `comment`, `issues`, and `tags`.
+To hide columns from the report, set the `hidden_columns` parameter, for example `https://www.quality-time.example.org/api/v3/report/<report_uuid>/pdf?hidden_columns=target,comment`.
+Possible options are `trend`, `status`, `measurement`, `target`, `source`, `comment`, `issues`, and `tags`.
```
Comment (fniessink) [inline docs/src/api.md]: These values are not up-to-date. At least `delta` is missing (my fault). 

```suggestion
Possible options are `trend`, `status`, `measurement`, `delta`, `target`, `source`, `comment`, `issues`, and `tags`.
```

File: docs/src/api.md [hunk 2]
```diff
@@ -114,17 +143,25 @@ curl --cookie-jar cookie.txt --request POST --header "Content-Type: application/
...
+The `report_uuid` is the unique identifier that *Quality-time* assigns to a report.
+It can be found by navigating to a report in the browser and looking for the `report_uuid` in the address bar.
+For example, when the URL in the browser's address bar is `https://quality-time.source.org/f1d0e056-2440-43bd-b640-f6753ccf4496?hidden_columns=comment`,
+the part between the last slash and the question mark is the `report_uuid`.
 
-The {index}`public key <Public key>` argument is optional. If no public key is provided, the public key of the exporting *Quality-time* instance is used for encrypting the source credentials. If the report needs to be imported in a different *Quality-time* instance, the public key of that instance should be provided. It can be obtained at `https://quality-time.destination.org/api/v3/public_key`. The exported JSON report can only be imported into the *Quality-time* whose public key has been used for the encryption of credentials during the export.
+The {index}`public key <Public key>` argument is optional.
```
Comment (fniessink) [inline docs/src/api.md]:
```suggestion
The {index}`public key <public key>` argument is optional.
```

**Review by fniessink — APPROVED** [review]

File: docs/src/api.md (hunk 1 above)
Comment (wkoot) [reply docs/src/api.md]: Hmm, I'm not sure how to get an exhaustive list. Searching on `hidden_columns` only yields test files, so I used `hiddenColumns.includes`. Is this complete? It seems that the value `unit` does not have a test case

Comment (fniessink) [reply docs/src/api.md]: To be addressed in #8237.

**Comment by wkoot** [comment]
Linked to #8237 for generating doc

**Pull request merged**